### PR TITLE
Fab Toolbar

### LINKED ISSUE #30: Fab Toolbar
[Angular Material Docs Link](https://material.angularjs.org/latest/demo/fabToolbar)

### PR DESCRIPTION
**Pull request by sreejithr**
- [x] Animated Floating Action Button toolbar.
- [x] Sample FAB toolbar in Example project.
- [x] Import in `src/index.js`
- [x] Animate elevation of button on touch.

**Known issues:**
1. The button needs to expand from 0x to 1x scale. On Android, scale of 0 is taken as 1. Hence, it first loads fully at 1x scale and THEN, animates from 0x to 1x (https://github.com/facebook/react-native/issues/6278).

Fixes #30

### CODE REVIEW

**Comment by ahmedlhanafy** [comment]
I will take a look at it tonight @sreejithr

File: src/components/FAB/FAB.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    }).start(onComplete);
+  }
+
+  expandButton = () => this.scaleButton(1);
+
+  shrinkButton = () => this.scaleButton(0);
+
+  _handlePressIn = () => {
+    Animated.timing(this.state.elevation, {
+      toValue: 12,
```
Comment (ahmedlhanafy) [inline src/components/FAB/FAB.js]: Refer to the `<Button/>` component to get the elevation values

File: src/components/FAB/FAB.js [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+      duration: 200,
+    }).start();
+  };
+
+  _handlePressOut = () => {
+    Animated.timing(this.state.elevation, {
+      toValue: 6,
```
Comment (ahmedlhanafy) [inline src/components/FAB/FAB.js]: Same with this one

File: src/components/FAB/FAB.js [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    return (
+        <View style={[ styles.container, style ]}>
+          <AnimatedPaper
+            elevation={this.state.elevation}
+            style={[
+              styles.buttonContainer,
+              { transform: [ { scale: this.state.buttonScale } ] },
+            ]}
+          >
+            <TouchableRipple
```
Comment (ahmedlhanafy) [inline src/components/FAB/FAB.js]: add `borderless` prop to it

File: example/src/FABExample.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    toolbarRightOpen: false,
+  };
+
+  dismissToolbar = () => {
+    this.setState({ toolbarLeftOpen: false, toolbarRightOpen: false });
+  }
+
+  render() {
+    return (
+        <TouchableHighlight
```
Comment (ahmedlhanafy) [inline example/src/FABExample.js]: This is not the right approach, We need to render the `Fab` into `Portal`

File: src/components/FAB/FABToolbar.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+  openToolbar() {
+    Animated.stagger(100, [
+      this.moveIn(this.props.direction),
+      this.scaleUp(),
+    ]).start(() => this.setState({ displayActions: true }));
+  }
+
+  closeToolbar(onComplete: () => void) {
+    const self = this;
```
Comment (ahmedlhanafy) [inline src/components/FAB/FABToolbar.js]: Why are you assigning this to a var here?

File: src/components/FAB/FAB.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+ * Floating Action Button (FAB) is a circled icon which typically floats
+ * above the UI. On press, it can either:
+ *
+ * 1. Invoke an Action.
+ * 2. Present user with a horizontal full-width strip of Actions.
+ * 3. Fling a set of related Actions in the form of circular buttons.
+ *
+ * https://material.io/guidelines/components/buttons-floating-action-button.html
+ */
+class FAB extends Component<DefaultProps, Props, State> {
```
Comment (satya164) [inline src/components/FAB/FAB.js]: FAB is short for Floating Action Button, so I think FAB is the correct name

File: src/components/FAB/FAB.js [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+     * The button needs to expand from 0x to 1x scale. On Android, scale of 0 is
+     * taken as 1. Hence, it first loads fully at 1x scale and THEN, animates from
+     * 0x to 1x (https://github.com/facebook/react-native#6278).
+     */
+    this.expandButton();
+  }
+
+  componentWillUnmount = () => this.shrinkButton();
+
+  scaleButton(toValue, onComplete) {
```
Comment (satya164) [inline src/components/FAB/FAB.js]: Also add `_`

**Comment by satya164** [comment]
I think we should not change to `Fab` because `FAB` is an acronym for Floating Action Button and not an actual word.

**Comment by geminiyellow** [comment]
@sreejithr @satya164  so this PR been killed? why dont you export it as other name or create a library?

**Pull request closed without merging**